core/basic_authorship/block_builder_factory.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_BASIC_AUTHORSHIP_BLOCK_BUILDER_FACTORY_HPP
+#define KAGOME_CORE_BASIC_AUTHORSHIP_BLOCK_BUILDER_FACTORY_HPP
+
+#include "basic_authorship/block_builder.hpp"
+#include "primitives/block_id.hpp"
+#include "runtime/block_builder_api.hpp"
+#include "runtime/core.hpp"
+
+namespace kagome::basic_authorship {
+
+  /**
+   * Creates new block builders
+   */
+  class BlockBuilderFactory {
+   public:
+    virtual ~BlockBuilderFactory() = default;
+
+    /**
+     * Prepares BlockBuilder for creating block on top of parent block and using
+     * provided digest
+     */
+    virtual outcome::result<std::unique_ptr<BlockBuilder>> create(
+        const primitives::BlockId &parent_id,
+        const primitives::Digest &inherent_digest) = 0;
```

Comment:
Can't the method be made `const`?

Reply:
internally it initialises the block, which is not const

Reply:
`const`-ness of the method means that it does not mutate any of the class variables, and this method doesn't, so it can be made `const`

Reply:
Mutating the r_core_, which is a shared_ptr, is not a mutating of runtime::Core object inside